stacks-core/src/utils.rs

```diff
@@ -212,4 +243,47 @@ mod tests {
 
 		assert_eq!(deserialized, expected_principal_data);
 	}
+
+	#[test]
+	fn should_principal_data_try_from_string() {
+		// addr = ST000000000000000000002AMW42H
+		let addr = StacksAddress::new(
+			AddressVersion::TestnetSingleSig,
+			Hash160Hasher::default(),
+		);
+		let principal_data: PrincipalData = PrincipalData::try_from(
+			"ST000000000000000000002AMW42H.helloworld".to_string(),
+		)
+		.unwrap();
+
+		assert_eq!(
+			principal_data,
+			PrincipalData::Contract(
+				StandardPrincipalData(addr.version(), addr.clone()),
+				ContractName::new("helloworld").unwrap(),
+			)
+		);
+	}
+
+	#[test]
+	fn should_fail_to_convert_invalid_string_to_principal_data() {
+		// try invalid address
+		let mut result =
+			PrincipalData::try_from("ST123.helloworld".to_string());
+
+		assert_eq!(
+			result.unwrap_err().to_string(),
+			"Could not crackford32 encode or decode: Invalid C32 address: ST123"
```

Comment:
Non Idiomatic way of asserting an error. Does something like this work? You can also try `assert_matches`.
```rust
assert_eq!(
    result
    Err(Error::Variant("Could not crackford32 encode or decode: Invalid C32 address: ST123"))
)
```

Reply:
nit, feel free to resolve it.

Reply:
I tried, but failed to find a solution other than Errors and `to_string` because `StacksError`does not implement PartialEq and my solutions failed so far.